Assignments/Sorting/InsertionSort.playground/Pages/Insertion Sort.xcplaygroundpage/Contents.swift

```diff
@@ -7,9 +7,18 @@ Note that there is already a `swap` function in the standard library. However, y
 
 */
 
-func insertionSort<T>(var array: [T], @noescape isOrderedBefore: (T, T) -> Bool) -> [T] {
-
-    return array
+func insertionSort<T>( array: [T], @noescape isOrderedBefore: (T, T) -> Bool) -> [T] {
+    var sortedArray : [T] = []
```

Comment:
The original function signature declared the `array` parameter as `var` which will give you a copy of the data that is mutable. Removing the `var` from the declaration still gets you a _copy_ of the array, but an immutable one. Here you're creating a _3rd_ copy of the data. This is fine for small arrays of strings or integers but it may be prohibitively inefficient with larger datasets.